app/controllers/my-projects/project/recommendations/add.js

```diff
@@ -25,58 +39,80 @@ export default class MyProjectsProjectRecommendationsAddController extends Contr
-  // to all actions
-  allActionsRecommendation = '';
-
-  recommendationOptions = EmberObject.create({
-    approved: RecommendationOption.create({
-      label: 'Approved',
-      code: 'approved',
-    }),
-    'approved-with-modifications-conditions': RecommendationOption.create({
-      label: 'Approved with Modifications/Conditions',
-      code: 'approved-with-modifications-conditions',
-    }),
-    disapproved: RecommendationOption.create({
-      label: 'Disapproved',
-      code: 'disapproved',
-    }),
-    'disapproved-with-modifications-conditions': RecommendationOption.create({
-      label: 'Disapproved with Modifications/Conditions',
-      code: 'disapproved-with-modifications-conditions',
-    }),
-    'not-available': RecommendationOption.create({
-      label: 'N/A',
-      code: 'not-available',
-    }),
-  });
-
-  @computed('recommendationOptions.{approved.actions.[],["approved-with-modifications-conditions"].actions.[],disapproved.actions.[],["disapproved-with-modifications-conditions"].actions.[],["not-available"].actions.[]}')
-  get allOptionsActions() {
-    const allActions = [
-      ...this.recommendationOptions.approved.actions,
-      ...this.recommendationOptions.get('approved-with-modifications-conditions').actions,
-      ...this.recommendationOptions.disapproved.actions,
-      ...this.recommendationOptions.get('disapproved-with-modifications-conditions').actions,
-      ...this.recommendationOptions.get('not-available').actions,
-    ];
-    return allActions;
-  }
+  // recommendation options for disposition.communityboardrecommendation,
+  // disposition.boroughboardrecommendation, disposition.boroughpresidentrecommendation
+  recOptions = [
+    'Approved',
+    'Approved with Modifications/Conditions',
+    'Disapproved',
+    'Disapproved with Modifications/Conditions',
+    'Waived',
+  ];
 
   @action
   setProp(property, newVal) {
     this.set(property, newVal);
   }
 
+  // For setting disposition recommendation, use the action
+  // "setDispositionRecByPartType" instead.
   @action
   updateDispositionAttr(disposition, attrName, newVal) {
     disposition.set(attrName, newVal);
   }
+
+  /**
+ * @param { Disposition } disposition
+ * @param { String } recommendation
+ * @param { String } participantType (optional)
+ * assigns the `recommendation` string to either the `
+ * TODO: Update this to rely on the disposition.participantType field
+ * when it is implemented in the ZAP-API.
+ */
+  @action
+  setDispositionRecByPartType (disposition, recommendation) {
+    const partType = this.participantType;
+    let recommendationFieldName = '';
+    if (partType === 'CB') {
+      recommendationFieldName = 'communityboardrecommendation';
+    } else if (partType === 'BB') {
+      recommendationFieldName = 'boroughboardrecommendation';
+    } else if (partType === 'BP') {
+      recommendationFieldName = 'boroughpresidentrecommendation';
+    } else {
```

Comment:
This is a lookup. Let's move this lookup to a constant above this controller class:

```
const PARTICIPANT_TYPE_FIELD_LOOKUP = {
  BB: 'boroughboardrecommendation',
  BP: 'boroughpresidentrecommendation',
};
```

Reply:
Then we can just say
```
const targetField = PARTICIPANT_TYPE_FIELD_LOOKUP[participantType];
disposition.set(targetField, recommendation);
```

Reply:
✅